cmd/rofl/mgmt.go

```diff
@@ -606,12 +606,15 @@ var (
 			if pubName != "" {
 				secretCfg.PublicName = pubName
 			}
+			var newSecrets []*buildRofl.SecretConfig
 			for _, sc := range deployment.Secrets {
-				if sc.Name == secretName {
-					cobra.CheckErr(fmt.Errorf("secret named '%s' already exists for deployment '%s'", secretName, roflCommon.DeploymentName))
+				if sc.Name != secretName {
+					newSecrets = append(newSecrets, sc)
```

Comment:
IMO building a new list of secrets doesn't make sense. I would either append or replace the item in the existing list:

```go
if sc.Name == secretName {
  common.CheckForceErr(fmt.Errorf("the secret named '%s' for deployment '%s' already exists", secretName, roflCommon.DeploymentName))
  deployment.Secrets[i] = &secretCfg
  replaced = true
  break
}
```

and then

```go
if !replaced {
  deployment.Secrets = append(deployment.Secrets, &secretCfg)
}
```